service/storage_service.cc

```diff
@@ -6277,13 +6277,15 @@ future<bool> storage_service::exec_tablet_update(service::group0_guard guard, st
 
 // Repair the tablets contain the tokens and wait for the repair to finish
 // This is used to run a manual repair requested by user from the restful API.
-future<std::unordered_map<sstring, sstring>> storage_service::add_repair_tablet_request(table_id table, std::variant<utils::chunked_vector<dht::token>, all_tokens_tag> tokens_variant) {
+future<std::unordered_map<sstring, sstring>> storage_service::add_repair_tablet_request(table_id table,
+        std::variant<utils::chunked_vector<dht::token>, all_tokens_tag> tokens_variant,
+        sstring hosts_filter, sstring dcs_filter) {
```

Comment:
Why not make these `std::unordered_set<hos_id>` and `std::unordered_set<sstring>` respectively? This way the interface already guarantees de-duplication and that host-ids are parseable.

Reply:
This way we would need to serialize it a moment later. It would probably be better from the interface perspective, but now we do not need to serialize the lists at all

Reply:
I think the interface clarity is worth it. Starting a repair is not on the hot path.